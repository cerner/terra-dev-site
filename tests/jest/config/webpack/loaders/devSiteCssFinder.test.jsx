import findCss from '../../../../../config/webpack/loaders/devSiteCssFinder';

describe('devSiteExample css finder', () => {
  it('should seperate a shortened css file path correctly', () => {
    const testText = 'Import from \'./pretendExampleCss.scss\'';
    const example = findCss(testText);
    expect(example).toMatchSnapshot();
  });

  it('should seperate a direct css file correctly', () => {
    const testText = 'Import from \'/Users/dm068655/Documents/ExternalR/terra-dev-site/tests/jest/config/webpack/loaders/pretendExampleCss.scss\'';
    const example = findCss(testText);
    expect(example).toMatchSnapshot();
  });

  it('should return undefined if there is no .css or .scss file declaration', () => {
    const testText = 'Import from \'./pretendExampleCss\'';
    const example = findCss(testText);
    expect(example).toMatchSnapshot();
  });

  it('should return undefined if there is not file declaration but the file name contains css', () => {
    const testText = 'Import from \'./pretendExampleCss\'';
    const example = findCss(testText);
    expect(example).toMatchSnapshot();
  });
});
