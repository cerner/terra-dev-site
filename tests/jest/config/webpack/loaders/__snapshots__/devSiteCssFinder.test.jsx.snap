// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`devSiteExample css finder should return undefined if there is no .css or .scss file declaration 1`] = `undefined`;

exports[`devSiteExample css finder should return undefined if there is not file declaration but the file name contains css 1`] = `undefined`;

exports[`devSiteExample css finder should seperate a direct css file correctly 1`] = `"/Users/dm068655/Documents/ExternalR/terra-dev-site/tests/jest/config/webpack/loaders/pretendExampleCss.scss"`;

exports[`devSiteExample css finder should seperate a shortened css file path correctly 1`] = `"./pretendExampleCss.scss"`;
